### DISCUSSION TITLE
Is there a plan to allow users to draw other shapes such as circles and triangles etc?

### DISCUSSION THREAD

**Discussion by Joe23232** [bevyengine/bevy#2472] [Q&A] [answered]
I know that Bevy currently allows the users to draw rectangles, but is there a plan when Bevy will allow the user to draw other shapes such as circles, triangles etc without needing third party extensions for Bevy such as Lyon?

**Comment by alice-i-cecile** [comment, marked as answer]
My understanding is Yes, Eventually:tm:.

IMO, this is likely to build off the work started in #1621. Stay tuned for a full RFC laying out the general shape of the vision on this and related features. As a sneak peek, my expectation is that we'll probably end up going:
1. Geometric primitives
2. Debug drawing (#1625), built off the geometric primitives, to actually render the geometric primitives
3. More customization options for the debug drawing
4. A bevy_shapes subcrate that uses these same geometric primitive options to create nice APIs for drawing other polygons and line types in 2D and 3D

**Comment by Joe23232** [comment]
Oh nice, is this for version `0.5` release?

**Comment by aevyrie** [comment]
No, 0.5 is getting wrapped up as we speak, I'd expect it to release next week. 

Primitives (#1621) will need an RFC first as it's pretty fundamental to the engine. I would expect primitives and debug to make it into 0.6.

**Comment by Joe23232** [comment]
Thanks mate

**Comment by alice-i-cecile** [comment]
Migrated to a [Discussion](https://github.com/bevyengine/bevy/discussions/2465).

**Reply by cart** [reply]
Just as a heads up to readers that are confused, we've opted to go with github's auto-migration feature instead. It works quite well!
